procrastinate/manager.py

```diff
@@ -51,37 +52,89 @@ async def defer_job_async(self, job: jobs.Job) -> jobs.Job:
         """
         # Make sure this code stays synchronized with .defer_job()
         try:
-            result = await self.connector.execute_query_one_async(
-                **self._defer_job_query_kwargs(job=job)
+            result = await self.connector.execute_query_all_async(
+                **self._defer_jobs_query_kwargs(jobs=[job])
             )
         except exceptions.UniqueViolation as exc:
             self._raise_already_enqueued(exc=exc, queueing_lock=job.queueing_lock)
 
-        return job.evolve(id=result["id"], status=jobs.Status.TODO.value)
+        return job.evolve(id=result[0]["id"], status=jobs_module.Status.TODO.value)
+
+    async def batch_defer_jobs_async(
+        self, jobs: list[jobs_module.Job]
+    ) -> list[jobs_module.Job]:
+        """
+        Add multiple jobs in their queue for later processing by a worker.
+
+        Parameters
+        ----------
+        jobs:
+            The jobs to defer
+
+        Returns
+        -------
+        :
+            A list of jobs with their id set.
+        """
+        # Make sure this code stays synchronized with .batch_defer_jobs()
+        try:
+            results = await self.connector.execute_query_all_async(
+                **self._defer_jobs_query_kwargs(jobs=jobs)
+            )
+        except exceptions.UniqueViolation as exc:
+            self._raise_already_enqueued(
+                exc=exc, queueing_lock=None
+            )  # TODO: fix queing_lock
+
+        return [
+            job.evolve(id=results[index]["id"], status=jobs_module.Status.TODO.value)
+            for index, job in enumerate(jobs)
+        ]
 
-    def defer_job(self, job: jobs.Job) -> jobs.Job:
+    def defer_job(self, job: jobs_module.Job) -> jobs_module.Job:
         """
         Sync version of `defer_job_async`.
         """
         try:
-            result = self.connector.get_sync_connector().execute_query_one(
-                **self._defer_job_query_kwargs(job=job)
+            results = self.connector.get_sync_connector().execute_query_all(
+                **self._defer_jobs_query_kwargs(jobs=[job])
             )
         except exceptions.UniqueViolation as exc:
             self._raise_already_enqueued(exc=exc, queueing_lock=job.queueing_lock)
 
-        return job.evolve(id=result["id"], status=jobs.Status.TODO.value)
+        return job.evolve(id=results[0]["id"], status=jobs_module.Status.TODO.value)
+
+    def batch_defer_jobs(self, jobs: list[jobs_module.Job]) -> list[jobs_module.Job]:
+        """
+        Sync version of `batch_defer_jobs_async`.
+        """
+        try:
+            results = self.connector.get_sync_connector().execute_query_all(
+                **self._defer_jobs_query_kwargs(jobs=jobs)
+            )
+        except exceptions.UniqueViolation as exc:
```

Comment:
Coverage tells me that L115-116 are not hit. I wonder why that is, because it should be tested by https://github.com/procrastinate-org/procrastinate/blob/95741d55563736ab02b3164066c5a65c7091b326/tests/integration/test_manager.py#L657